internal/inputs/inputs_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inputs
+
+import (
+	"errors"
+	"reflect"
+	"sync"
+	"testing"
+)
+
+type inputMock struct {
+	getIDFn          func() string
+	getDescriptionFn func() string
+	getDataFn        func(clusterID string) (interface{}, error)
+	closeFn          func() error
+}
+
+func (m inputMock) GetID() string {
+	return m.getIDFn()
+}
+
+func (m inputMock) GetDescription() string {
+	return m.getDescriptionFn()
+}
+
+func (m inputMock) GetData(clusterID string) (interface{}, error) {
+	return m.getDataFn(clusterID)
+}
+
+func (m inputMock) Close() error {
+	return m.closeFn()
+}
+
+func TestGetAllInputsData(t *testing.T) {
+	clusterIDs := []string{"cluster-one", "cluster-two", "cluster-three", "cluster-four"}
+	inputs := []Input{
+		&inputMock{getIDFn: func() string { return "gke-api" }, getDataFn: func(clusterID string) (interface{}, error) { return "data", nil }},
+		&inputMock{getIDFn: func() string { return "kube-api" }, getDataFn: func(clusterID string) (interface{}, error) { return "data", nil }},
+		&inputMock{getIDFn: func() string { return "metrics-api" }, getDataFn: func(clusterID string) (interface{}, error) { return "data", nil }},
+		&inputMock{getIDFn: func() string { return "bad-api" }, getDataFn: func(clusterID string) (interface{}, error) { return nil, errors.New("error") }},
+	}
+	results, errors := GetAllInputsData(inputs, clusterIDs)
+	if len(results) != len(clusterIDs) {
+		t.Fatalf("number of results = %v; want %v", len(results), len(clusterIDs))
+	}
+	if len(errors) != len(clusterIDs) {
```

Comment:
shouldn't we expect empty error list?

Reply:
The test `inputs[]` contain one mock input ("bad-api") that produces an error on `getData()`.
Therefore, for each cluster, there will one error from this input. The idea was to cover error results processing as well.